### PR TITLE
feat!: add v2 conversion command

### PR DESCRIPTION
**Pull request by vasco-santos**
Per feedback on https://github.com/web3-storage/w3filecoin-infra/pull/80#issuecomment-1823011924 adds conversion command.

Note that needed to upgrade libs that generate PieceCIDv2 now, but will also get a separate PR for it individually to go first. So needs https://github.com/web3-storage/pieces/pull/6

```sh
➜  pieces git:(feat/add-v2-conversion-command) ✗ node cli.js v2 baga6ea4seaqhmw7z7q3jypdr54xaluhzdn6syn7ovovvjpaqul2qqenhmg43wii --log-size 35
bafkzcibcaaphmw7z7q3jypdr54xaluhzdn6syn7ovovvjpaqul2qqenhmg43wii
➜  pieces git:(feat/add-v2-conversion-command) ✗ node cli.js v2 baga6ea4seaqhmw7z7q3jypdr54xaluhzdn6syn7ovovvjpaqul2qqenhmg43wii --height 30  
bafkzcibcaaphmw7z7q3jypdr54xaluhzdn6syn7ovovvjpaqul2qqenhmg43wii
```

### CODE REVIEW

File: src/v2.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  if (!height && !logSize) {
+    return exit(400, 'needs either height or log2 size of the Piece')
+  }
+
+  /** @type {import('@web3-storage/data-segment').LegacyPieceLink} */
+  let pieceCidV1
+  try {
+    pieceCidV1 = parseLink(cidStr)
+  } catch {
+    throw new Error(`PieceCIDv1 received ${cidStr} is not a valid CID`)
```
Comment (alanshaw) [inline src/v2.js]: Use `exit` here also?

File: src/v2.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * `v2` command converts PieceCIDv1 to PieceCIDv2 with its height embedded.
+ *
+ * @param {string} cidStr
+ * @param {Options} opts
+ */
+export default async function v2 (cidStr, opts) {
+  const height = opts.height ? Number(opts.height) : undefined
+  const logSize = opts['log-size'] ? Number(opts['log-size']) : undefined
```
Comment (alanshaw) [inline src/v2.js]: Should we be using `parseInt` here? Or at least checking we got a positive integer?

File: src/v2.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  if (!height && !logSize) {
```
Comment (alanshaw) [inline src/v2.js:20]: Can `height` or `logSize` be zero? If yes then this should be a null/undefined check.

File: src/v2.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const piece = convertPieceCidV1toPieceCidV2(
+      pieceCidV1,
+      height
+    )
+
+    console.log(piece.toString())
+  } else {
+    const piece = convertPieceCidV1toPieceCidV2(
+      pieceCidV1,
+      log2PieceSizeToHeight(Number(logSize))
```
Comment (alanshaw) [inline src/v2.js]: Don't need to cast again here right?

```suggestion
      log2PieceSizeToHeight(logSize)
```

File: cli.js [hunk 5]
```diff
@@ -20,4 +21,9 @@ cli.command('verify [car] [cid]', 'check the car piece cid is correct')
 cli.command('find <piece>', 'resolve url for piece cid')
   .action(find)
 
+cli.command('v2 [pieceCid]', 'convert PieceCIDv1 to PieceCIDv2')
```
Comment (alanshaw) [inline cli.js]: It is required right?

```suggestion
cli.command('v2 <pieceCid>', 'convert PieceCIDv1 to PieceCIDv2')
```

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: src/v2.js (hunk 3 above)
Comment (vasco-santos) [reply src/v2.js:20]: by the minimum size of a piece no

**Review by alanshaw — APPROVED** [review]

**Pull request merged**